src/LoadingError.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Class LoadingError.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Service;
+
+/**
+ * Client-side app loading error markup and styles.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.1.3
+ * @internal
+ */
+final class LoadingError implements Service {
+
+	/**
+	 * Render error message along with necessary styles.
+	 */
+	public function render() {
+		?>
+		<div id="amp-pre-loading-spinner" class="amp-spinner-container">
+			<span class="amp-loading-spinner">
+				<span class="screen-reader-text">
+					<?php esc_html_e( 'Loading', 'amp' ); ?>
+				</span>
+			</span>
+		</div>
+
+		<div id="amp-loading-failure" class="error-screen-container">
+			<div class="error-screen components-panel">
+				<h1>
+					<?php esc_html_e( 'Something went wrong.', 'amp' ); ?>
+				</h1>
+
+				<?php
+				printf(
+					'<p class="amp-loading-failure-script">%s</p>',
+					wp_kses(
+						sprintf(
+							/* translators: %s is the AMP support forum URL. */
+							__( 'Check your connection and open your browser console to see if there are any error messages. You may share them on the <a href="%s" target="_blank" rel="noreferrer noopener">support forum</a>.', 'amp' ),
+							esc_url( __( 'https://wordpress.org/support/plugin/amp/', 'amp' ) )
+						),
+						[
+							'a' => [
+								'href'   => true,
+								'target' => true,
+								'rel'    => true,
+							],
+						]
+					)
+				);
+				?>
+
+				<noscript>
+					<p class="amp-loading-failure-noscript"><?php esc_html_e( 'You must have JavaScript enabled to use this page.', 'amp' ); ?></p>
+				</noscript>
+			</div>
+		</div>
+		<style>
+			#amp-loading-failure {
+				visibility: hidden;
+				animation: amp-wp-show-element 30s steps(1, end) 0s 1 normal both;
+			}
+
+			#amp-pre-loading-spinner {
+				visibility: visible;
+				animation: amp-wp-hide-element 30s steps(1, end) 0s 1 normal both; /* This could probably reuse amp-wp-show-element if reversed. */
```

Comment:
I'm not sure how a slow connection would come into play here, as the page would have already loaded and the necessary JS would have either executed or not by then. A 30s wait time seems to be overkill here.

Reply:
I was testing with Fast 3G and the JS files were taking over 30 seconds to load. Granted, I was testing with the development versions, so it was 8MB (😱). I'll test again with the production version instead and adjust the timeout down.

Reply:
With Fast 3G, the settings page takes 10 seconds to load. With Slow 3G it takes 36 seconds to load. This is with the production build, where `amp-settings.js` is 875KB. So I think 30 seconds is not actually far off.

Reply:
Ah I see. That makes sense.

Reply:
What an ingenious solution!